src/applications/renderer/src/components/legend/component.js

```diff
@@ -45,12 +47,17 @@ const Legend = ({
         alias: option.label !== option.value ? option.label : undefined,
       };
 
-    if (isPie) {
-      patchConfiguration({ category: newOption });
+    if (isPie || isDonut) {
+      patchConfiguration({
+        category: newOption,
+        color: newOption
+      });
     } else {
-      patchConfiguration({ color: newOption });
+      patchConfiguration({
+        category: newOption
+      });
```

Comment:
This is an error. When the chart is not a pie or a donut, the field in the legend must update the colour setting.

You can test that this is broken by creating a bar chart with “Country” on the X axis (category), “Gross GDP” on the Y (value), and “Country” again in the legend field. What should happen is that each of the bars gets a different colour. What's happening with this change is that the field does nothing.